frontend/src/router.ts

```diff
@@ -180,46 +181,84 @@ export const router = new Router({
   ]
 });
 
+export const hasClaim = (
+  requiredClaims: { [key: string]: any },
+  userClaims: { [key: string]: any }
+) => {
+  return Object.entries(requiredClaims).some(
+    ([claimName, claimValue]) => userClaims[claimName] === claimValue
+  );
+};
+
+export const filterRoutesByClaims = (
+  routes: RouteConfig[] = [],
+  userClaims: { [key: string]: any },
+  requireParentClaims: boolean | { [key: string]: any } = false
+) => {
+  return routes.reduce(
+    (filteredRoutes, route) => {
+      const requireClaims = _.get(route, "meta.auth.requireClaims", false);
+
+      if (route.meta.activator) {
+        const childRoutes = filterRoutesByClaims(
+          route.children,
+          userClaims,
+          requireClaims || requireParentClaims
+        );
+        childRoutes.length &&
+          filteredRoutes.push({ ...route, children: childRoutes });
+      } else {
+        if (requireClaims) {
+          hasClaim(requireClaims, userClaims) && filteredRoutes.push(route);
+        } else {
+          if (typeof requireParentClaims === "object") {
+            hasClaim(requireParentClaims, userClaims) &&
```

Comment:
Why don't you put this check directly into `if` condition? I guess effect is the same, but using `&&` makes understanding the code harder. And above also.

Reply:
If it was in the `if` condition, I think it would have unintended side effects. For example if `requireClaims` exists but the user doesn't have any claims it would go to else and check parent but that is not the desired effect.

Reply:
OK, let it be as you implemented.